util/topgen/intermodule.py

```diff
@@ -146,9 +187,121 @@ def find_intermodule_signal(sig_list, m_name, s_name) -> Dict:
         x for x in sig_list if x["name"] == s_name and x["inst_name"] == m_name
     ]
 
-    assert len(
-        filtered
-    ) == 1, "Error on finding the inter module signal {m_name}.{s_name}".format(
-        m_name=m_name, s_name=s_name)
+    return filtered[0] if len(filtered) == 1 else None
+
+
+## Validation
+def check_intermodule(topcfg: Dict, prefix: str) -> int:
+    if "inter_module" not in topcfg:
+        return 0
+
+    total_error = 0
+
+    for req, rsps in topcfg["inter_module"].items():
+        error = 0
+        # checking the key, value are in correct format
+        # Allowed format
+        #   1. module.signal
+        #   2. module.signal[index]
```

Comment:
could you provide examples for each of these cases? i feel like that would be easier to digest to a reader.. or more likely..someone who did something wrong and is trying to figure out why :)

Reply:
OK I will revise the code to have more examples there.
Also the missing part in this PR is that updating the docs.
Comportability spec and topgen manual page need to be revised to have `inter_signal_list` and `inter_module` fields.